source/misc/bash_startup.in

```diff
@@ -1,257 +1,95 @@
-#!/bin/bash
-# This is based on "preexec.bash" but is customized for iTerm2.
+# This is inspired by "preexec.bash" but is customized for iTerm2. Sourced, so
+# no shebang. This module requires 2 bash features you must not otherwise be
+# using: the DEBUG trap and the PROMPT_COMMAND variable.
 
-# Note: this module requires 2 bash features which you must not otherwise be
-# using: the "DEBUG" trap, and the "PROMPT_COMMAND" variable.  iterm2_preexec_install
-# will override these and if you override one or the other this _will_ break.
+osc='\033]'; cso='\007' # Beginning, ending escape sequences.
 
-# This is known to support bash3, as well as *mostly* support bash2.05b.  It
-# has been tested with the default shells on MacOS X 10.4 "Tiger", Ubuntu 5.10
-# "Breezy Badger", Ubuntu 6.06 "Dapper Drake", and Ubuntu 6.10 "Edgy Eft".
+# tmux / screen are not supported; I think this is the theoretical tmux hack?
+[ $TERM = screen ] && osc='\033Ptmux;\033\033]' && cso='\007\033\\'
 
+orig_ps1=$PS1; prev_ps1=$PS1 # orig is the user-set PS1; prev is our last PS1.
+preexec_interactive_mode= # If set: "just executed prompt; waiting for input."
 
-# Copy screen-run variables from the remote host, if they're available.
+[ -z "$iterm2_hostname" ] && iterm2_hostname=$(uname -n) # hostname -f == slow
```

Comment:
On most systems it should be identical, but `uname` is never supposed to query an external DNS server, I believe (which is why `hostname` can be slow). Potentially there might be an issue if there's a proper DNS A record but not a matching PTR record, and the server sets another, default hostname. Not sure though.